src/renderer/qssWidget/js/qssOpenUsbWidget.js

```diff
@@ -0,0 +1,350 @@
+                args: [
+                    "{qssWidget}.container",
+                    "{that}.dom.menuControlsWrapper",
+                    "{that}.dom.heightListenerContainer"
+                ]
+            },
+            close: {
+                funcName: "gpii.qssWidget.openUSB.close",
+                args: [
+                    "{that}",
+                    "{closeTimer}",
+                    "{arguments}.0" // keyboardEvent
+                ]
+            },
+            showNotification: {
+                func: "{that}.events.onNotificationRequired.fire",
+                args: [{
+                    description: "{arguments}.0",
+                    closeOnBlur: "{that}.model.closeOnBlur"
+                }]
+            }
+        },
+        events: {
+            onHeightChanged: null
+        }
+    });
+
+    /**
+     * Invoked whenever the user changes the value of the given setting. Schedules that
+     * the widget should be closed in `closeDelay` milliseconds.
+     * @param {Component} that - The `gpii.qssWidget.menu` instance.
+     * @param {Component} closeTimer - An instance of `gpii.app.timer` used for closing
+     * the widget with a delay.
+     * @param {KeyboardEvent} keyboardEvent - The keyboard event (if any) that led to the
+     * change in the setting's value.
+     */
+    gpii.qssWidget.openUSB.close = function (that, closeTimer, keyboardEvent) {
+        that.closingKeyboardEvent = keyboardEvent;
+        closeTimer.start(that.options.closeDelay);
+    };
+
+    /**
+     * Adds a listener which stops the close timer in case the dialog was closed before
+     * the timer has finished as a result of another user interaction (e.g. clicking
+     * outside of the widget's window or clicking on a different QSS button which opens
+     * the QSS widget again).
+     * @param {Component} closeTimer - An instance of `gpii.app.timer` used for closing
+     * the widget with a delay.
+     */
+    gpii.qssWidget.openUSB.addVisibilityChangeListener = function (closeTimer) {
+        $(document).on("visibilityChange.qssOpenUSBWidget", function () {
+            if (document.visibilityState === "hidden") {
+                closeTimer.clear();
+            }
+        });
+    };
+
+    /**
+     * Removes the listener for clearing the `closeTimer`. Useful when the component is
+     * destroyed.
+     */
+    gpii.qssWidget.openUSB.removeVisibilityChangeListener = function () {
+        $(document).off("visibilityChange.qssOpenUSBWidget");
+    };
+
+    /**
+     * Creates an array of possible values for the given setting to be provided to the
+     * `gpii.psp.repeater` instance.
+     * @param {Object} setting - the current setting for the QSS menu.
+     * @return {Object[]} - An array of key-value pairs describing the key and the value
+     * (the visible name) of the setting.
+     */
+    gpii.qssWidget.openUSB.getRepeaterItems = function (setting) {
+        var schema = setting.schema || {},
```

Comment:
There doesn't seem to be a setting or a schema associated with the openUSB functionality - what is this function used for?